packages/zkdb/src/merkle-tree/merkle-tree-stream.ts

```diff
@@ -10,22 +10,26 @@ export type TMerkleTreeData = {
 export class MerkleTreeReadStream extends Readable {
   private merkleTreeData: TMerkleTreeData;
   private currentIdx: number = 0;
+  private batchSize: number;
 
-  constructor(data: TMerkleTreeData) {
+  constructor(data: TMerkleTreeData, batchSize = 10) {
     super({ objectMode: true });
     this.merkleTreeData = data;
+    this.batchSize = batchSize;
   }
 
   _read() {
-    let more = true;
-    while (more && this.currentIdx < this.merkleTreeData.nodes.length) {
+    const remaining = this.merkleTreeData.nodes.length - this.currentIdx;
+    const limit = Math.min(this.batchSize, remaining);
+
+    for (let i = 0; i < limit; i++) {
       const node = this.merkleTreeData.nodes[this.currentIdx++];
-      more = this.push(this.serializeNode(node));
-    }
-    if (!more) {
-      this.currentIdx--;
-      return;
+      const serializedNode = this.serializeNode(node);
+      if (!this.push(serializedNode)) {
+        return;
+      }
     }
+
     if (this.currentIdx === this.merkleTreeData.nodes.length) {
       const heightJSON = JSON.stringify({ height: this.merkleTreeData.height });
       this.push(Buffer.from(heightJSON, 'utf8'));
```

Comment:
I got some problem with `UTF-8` encoding since the value was stored in `ISO`. Btw, `buffer` won't work in browser.

Reply:
Merkle tree primarily consists of numbers. I guess, we can proceed only with a single type of encoding, just enforce write and read stream to store data in `UTF-8`